functions/src/modules/SQR.functions.ts

```diff
@@ -248,3 +251,73 @@ export const processSubmissions = functions.database.ref('/webforms/sqr/{submiss
     
     return 1;
 });
+
+
+/////////////////////////////////////////////////
+//          Import Submission and Allotments from a Spreadsheet(Http Triggered)
+//
+//      1. Parses a google spreadsheet
+//      2. Looks for two sheets --> Allotments & Submissions
+//      3. Loads their data into the equivalent Firebase database paths
+/////////////////////////////////////////////////
+export const importSpreadSheetData = functions.https.onRequest( async (req, res) => {
+    const auth = await google.auth.getClient({
+        scopes: ["https://www.googleapis.com/auth/spreadsheets"]
+    });
+    
+    const spreadsheetUrl = 'https://docs.google.com/spreadsheets/d/1e0rYguPua_0Rtd--uNLKkGotNw-CPM44TlBS5hthdfQ/edit#gid=477553865';
+    // extracting the spreadsheet ID from the url
+    const spreadsheetId = new RegExp("/spreadsheets/d/([a-zA-Z0-9-_]+)").exec(spreadsheetUrl)[1];
+    const spreadsheet = new GoogleSpreadsheet(spreadsheetId);
+
+    const token = await auth.getAccessToken();
+    spreadsheet.setAuthToken(token);
+
+    spreadsheet.getInfo((err, data) => {
+        if (!err) {
+            data.worksheets.forEach(worksheet => {
+                if(worksheet.title === "Submissions")
+                    worksheet.getRows({ }, (err, rows) => {
+                        rows.forEach(row => {
+                            let documentID = row['submissionserial'];
+                            db.ref(`/sqr/submissions/s${documentID}`)
+                                .set({
+                                    devotee: {
+                                        name: row['devotee'],
+                                        emailAddress: row['email'],
+                                    },
```

Comment:
Here is something wrong. Firstly, still `devotee` is here. Secondly, column names are not matching the [Submissions sheet](https://docs.google.com/spreadsheets/d/1e0rYguPua_0Rtd--uNLKkGotNw-CPM44TlBS5hthdfQ/edit#gid=729796178)

Reply:
1. I worked on this before you told us about restructuring the data ( I was just waiting for your comments here before moving on with the restructures data )
2. Column names are converted by the Node library ( lower-cased, and then spaces are removed ) ... I inspected the object first before using the column titles.

Reply:
![image](https://user-images.githubusercontent.com/1976151/48433179-2c022100-e77f-11e8-814c-df0f181209cd.png)
![image](https://user-images.githubusercontent.com/1976151/48433217-3de3c400-e77f-11e8-953d-608340914ce6.png)